File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        vowel = set(['a', 'e', 'i', 'o', 'u'])    
        
        ret = 0
        cur = 0
        left = 0

        for i in range(k):
            if s[i] in vowel:
                cur += 1

        ret = cur

        for i in range(k, len(s)):
            if s[left] in vowel:
                cur -= 1
            if s[i] in vowel:
                cur += 1
            left += 1    
            
            ret = max(ret, cur)

        return ret


        

        

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.java
import java.util.*;

class Solution {
    public int maxVowels(String s, int k) {
        // String이 아닌 Character 유의, Character는 작은 따음표 사용
        final HashSet<Character> vowel = new HashSet<>(Arrays.asList('a', 'e', 'i', 'o', 'u'));
        int cur = 0;
        int ret = 0;    
        int start = 0;

        for (int i = 0; i < k; i++) {
            if (vowel.contains(s.charAt(i))) {
                cur++;
                ret++;
            }
        }

        for (int i = k; i < s.length(); i++) {
            if (vowel.contains(s.charAt(start))) {
                cur--;
            }
            if (vowel.contains(s.charAt(i))) {
                cur++;
            }

            ret = Math.max(ret, cur);

            start++;
        }    

        return ret;
    }
}